src/services/functionAppService.ts

```diff
@@ -43,63 +44,45 @@ export class FunctionAppService extends BaseService {
     return response.data.value;
   }
 
-  public async deleteFunction(functionName) {
-    this.serverless.cli.log(`-> Deleting function: ${functionName}`);
-    return await this.webClient.webApps.deleteFunction(this.resourceGroup, this.serviceName, functionName);
-  }
-
-  public async syncTriggers(functionApp) {
-    this.serverless.cli.log('Syncing function triggers');
-
-    const syncTriggersUrl = `${this.baseUrl}${functionApp.id}/syncfunctiontriggers?api-version=2016-08-01`;
-    await this.sendApiRequest('POST', syncTriggersUrl);
-  }
+  async zipDeploy(functionApp) {
+    const functionAppName = functionApp.name;
+    this.serverless.cli.log(`Deploying zip file to function app: ${functionAppName}`);
 
-  public async cleanUp(functionApp) {
-    this.serverless.cli.log('Cleaning up existing functions');
-    const deleteTasks = [];
-
-    const serviceFunctions = this.serverless.service.getAllFunctions();
-    const deployedFunctions = await this.listFunctions(functionApp);
-
-    deployedFunctions.forEach((func) => {
-      if (serviceFunctions.includes(func.name)) {
-        this.serverless.cli.log(`-> Deleting function '${func.name}'`);
-        deleteTasks.push(this.deleteFunction(func.name));
-      }
-    });
-
-    return await Promise.all(deleteTasks);
-  }
-
-  public async listFunctions(functionApp) {
-    const getTokenUrl = `${this.baseUrl}${functionApp.id}/functions?api-version=2016-08-01`;
-    const response = await this.sendApiRequest('GET', getTokenUrl);
-
-    return response.data.value || [];
-  }
-
-  public async uploadFunctions(functionApp): Promise<any> {
-    this.serverless.cli.log('Creating azure functions');
-
-    const scmDomain = functionApp.enabledHostNames[0];
+    // Upload function artifact if it exists, otherwise the full service is handled in 'uploadFunctions' method
     const functionZipFile = this.serverless.service['artifact'];
+    if (functionZipFile) {
+      this.serverless.cli.log(`-> Uploading ${functionZipFile}`);
+
+      const uploadUrl = `https://${functionAppName}${constants.scmDomain}${constants.scmZipDeployApiPath}`;
```

Comment:
I don't think these constants should be used.  In theory this should be the only function that should use this URL.  Also - the `scmDomain` is not a constant and will change based on the configuration of the function app.

Reply:
`scmDomain` is coming from `config.ts`, how will that change with the function app?